src/python/pants/backend/terraform/testutil.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from dataclasses import dataclass
+from pathlib import Path
+from textwrap import dedent  # noqa: PNT20
+
+import pytest
+
+from pants.backend.terraform import dependencies, dependency_inference, tool
+from pants.backend.terraform.dependencies import TerraformInitRequest, TerraformInitResponse
+from pants.backend.terraform.goals.deploy import DeployTerraformFieldSet
+from pants.backend.terraform.goals.deploy import rules as terraform_deploy_rules
+from pants.backend.terraform.target_types import TerraformDeploymentTarget, TerraformModuleTarget
+from pants.core.goals import deploy
+from pants.core.goals.deploy import DeployProcess
+from pants.core.register import rules as core_rules
+from pants.core.util_rules import source_files
+from pants.engine import process
+from pants.engine.internals.native_engine import Address
+from pants.engine.rules import QueryRule
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner_with_auto_approve() -> RuleRunner:
```

Comment:
This is the first time a pytest fixture returning a `RuleRunner` is named something else than `rule_runner`. In this package, each test module has its own `rule_runner` fixture that is local to the module. This one, however, is being used by two independent test modules. Using `conftest.py` is not common and arguably not justified for a single fixture. Keeping the fixture with the same name may be slightly confusing (as, again, it's not local to a particular module) so to differentiate between them I've tweaked the name.

Reply:
I think giving these descriptive names is a good practice for these shared fixtures. We rebind them later and give them a more concise name closer to where they're used.